psa-crypto/src/types/key_derivation_operation.rs

```diff
@@ -0,0 +1,265 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+//! # PSA Operation types
+
+use super::algorithm::{Hash, KeyDerivation};
+use super::key::Id;
+#[cfg(feature = "interface")]
+use super::status::{Error, Result, Status};
+#[cfg(feature = "interface")]
+use core::convert::{From, TryFrom};
+
+/// Key derivation operation for deriving keys from existing sources
+#[derive(Debug, Clone)]
+pub struct KeyDerivationOperation<'a> {
+    /// Key derivation algorithm and inputs
+    pub alg: KeyDerivationWithInputs<'a>,
+    /// Maximum capacity of a key derivation operation
+    pub capacity: Option<usize>,
+}
+
+/// Wrapper around KeyDerivation to enforce correct `Input`s
+#[derive(Debug, Clone)]
+pub enum KeyDerivationWithInputs<'a> {
+    /// HKDF algorithm.
+    Hkdf {
+        /// A hash algorithm to use
+        hash_alg: Hash,
```

Comment:
Where is the hash alg actually used?

Reply:
When converting to `FullLengthMac::Hkdf` as it takes a hash algorithm (as does the [C macro](https://armmbed.github.io/mbed-crypto/html/api/ops/kdf.html?#c.PSA_ALG_HKDF) to build a Hkdf derivation algorithm).